tests/test_data.py

```diff
@@ -32,18 +32,21 @@ def test_load_data_with_labels(self):
         self.assertEqual(len(data), 15)
 
     def test_load_data_without_labels(self):
-        """ Tests load_data with an unlabeled dataset.
+        """ Tests load_data with an unlabeled sparse and dense dataset.
         """
-        x, y = make_blobs(n_samples=1500)
-        data = load_data(x=x, subset_size=100)
+        x = np.random.random((100, 2))
+        dataset = load_data(x=x, subset_size=10)
 
-        read_x = np.empty((0, x.shape[1]))
+        self.assertTrue((dataset.samples == x).all())
+        self.assertEqual(len(dataset), 10)
+        self.assertFalse(dataset.sparse)
 
-        for subset in data:
-            read_x = np.concatenate((read_x, subset.samples))
+        x = csr_matrix(x)
+        dataset = load_data(x=x, subset_size=10)
 
-        self.assertTrue((read_x == x).all())
-        self.assertEqual(len(data), 15)
+        self.assertTrue(np.allclose(dataset.samples.toarray(), x.toarray()))
```

Comment:
Why do you use allclose here and == above? Isn't it better to use np.array_equal?

Reply:
it was because when the type was not preserved, the casting produced numerical problems and not equal results. Fixed now that the type is preserved, gonna change it.